### PR TITLE
DSGVO compliance

### PR DESCRIPTION
**Pull request by btw04**
This PR aims to implement all necessary DSGVO features. 
- Allowing users to download all user data
- Deleting all user data on request

### CODE REVIEW

File: nextcloud/nextcloud_manager.py [hunk 1]
```diff
@@ -71,10 +75,58 @@ def file_exists(nc_path: PathLike[bytes] | str) -> bool:
...
+            path = os.path.basename(file.user_path)
+            if target_pattern.match(path):
+                return file.user_path
+    except NextcloudException:
+        return None
+    return None
+
+
+@_check_initialized
+def generate_observation_path(observation: AbstractObservation) -> str:
```
Comment (WickAJ) [inline nextcloud/nextcloud_manager.py:106]: Why is offset no parameter anymore even though you added the get_observation_file?

File: dsgvo/templates/dsgvo/index.html [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                    alert("User data has been successfully deleted.");
+                    window.location.href = '/';
+                } else {
+                    alert("An error occurred while deleting the data.");
+                }
+            });
+        }
+    });
+
+    </script>
```
Comment (WickAJ) [inline dsgvo/templates/dsgvo/index.html:161]: Text on the website is not readable (see image)
image

Comment (WickAJ) [inline dsgvo/templates/dsgvo/index.html:161]: It is also currently possible to delete the admin user, which imo should not be possible or have an additional layer of interaction (e.g. making another user admin) to ensure there is always at least one admin.

Comment (WickAJ) [inline dsgvo/templates/dsgvo/index.html:161]: When downloading the information about the admin user, the user_permissions stay empty. In my understanding this is not the expected behavior.

**Review by WickAJ — CHANGES_REQUESTED** [review]

File: dsgvo/templates/dsgvo/index.html (hunk 2 above)
Comment (btw04) [reply dsgvo/templates/dsgvo/index.html:161]: It is expected behavior, as we currently do not use "user_permissions". Instead, admins are defined by "is_superuser"

Comment (btw04) [reply dsgvo/templates/dsgvo/index.html:161]: Graphic-Design-Is-My-Passion

Comment (btw04) [reply dsgvo/templates/dsgvo/index.html:161]: It is noteworthy that these buttons are very much a placeholder. They _are_ visible in some browser, but I will improve visibility

Comment (btw04) [reply dsgvo/templates/dsgvo/index.html:161]: I disagree. The deletion is self-initialized. We should add a feature for admins to remove other users (and admins?) but this self-deletion should (is required to!) always work. 

In the worst case it is still pretty easy to set up a new admin user for us using the shell

**Review by WickAJ — APPROVED** [review]
LGTM

**Pull request merged**